control_plane/attachment_service/src/service.rs

```diff
@@ -336,6 +342,32 @@ impl Service {
         tracing::info!("Startup complete, spawned {reconcile_tasks} reconciliation tasks ({shard_count} shards total)");
     }
 
+    async fn initial_heartbeat_round(&self) -> HashMap<NodeId, PageserverUtilization> {
```

Comment:
We'll now be doing two API calls to pageservers at startup (list location_conf, and get utilization) -- that's fine, but I wonder if  we should skip the utilization GET if the location_conf list failed?  We may have already waited some seconds for a timeout.  This function could take a list of nodes to not bother heartbeating, based on who didn't respond to their location_conf.

Reply:
https://github.com/neondatabase/neon/pull/7092/commits/e603c0c3489ff462f41f0d6c64fc1a5a42ee9394